### PR TITLE
Temp: just for review PR65 rebased on LLVM 16

### PR DESCRIPTION
**Pull request by gerekon**
just for review [PR65](https://github.com/espressif/llvm-project/pull/65) rebased on LLVM 16

### CODE REVIEW

**Comment by gerekon** [comment]
@maciej-czekaj Could you review your changes ported onto LLVM 16?

@andreisfr FYI

File: llvm/lib/Target/Xtensa/XtensaBRegFixupPass.cpp [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      */
+    MachineOperand Breg = MI->getOperand(0);
+    MachineOperand Dst1 = MI->getOperand(1);
+    MachineOperand Dst2 = MI->getOperand(2);
+    MachineOperand Src = MI->getOperand(3);
+    DebugLoc DL = MI->getDebugLoc();
+    unsigned RegNo = Breg.getReg().id() - RegBase;
+    int64_t Mask = 0xffff & (~(1 << RegNo));
+
+    BuildMI(MBB, MI, DL, TII.get(Xtensa::RSR)).add(Dst1).addReg(Xtensa::BREG);
```
Comment (maciej-czekaj) [inline llvm/lib/Target/Xtensa/XtensaBRegFixupPass.cpp:118]: Should be `MachineInstrBuilder MIB = ...`
Required by LLVM_DEBUG statement below.

**Review by maciej-czekaj — APPROVED** [review]
Te patches looks good wit one small fix (in the comment).
Thanks for the rebase!

**Comment by gerekon** [comment]
Closed in favor of https://github.com/espressif/llvm-project/pull/80

**Pull request closed without merging**